### ISSUE TITLE
Why doesn't yolo-nas plot anything when using GPU, but plot when using CPU even though I already installed torch in cuda?

### ISSUE THREAD

**Issue by RamesChan** [Deci-AI/super-gradients#1851]
### 💡 Your Question

How can I fix this issue? I want to use gpu to run yolo-nas !!

When using "cuda", this code runs normally, but the results displayed are the same image, without plotting.
```
from super_gradients.training import models

yolo_nas_pose = models.get("yolo_nas_l", pretrained_weights="coco")

yolo_nas_pose.to('cuda').predict("3.jpg").show()
```
![Screenshot 2024-02-21 120239](https://github.com/Deci-AI/super-gradients/assets/128948774/fae3ebc8-b3fb-469d-92cd-b618ceb98a12)

But when I use cpu the results displayed are plot bounding box in image
```
yolo_nas_pose.to('cpu').predict("3.jpg").show()
```
![Screenshot 2024-02-21 120226](https://github.com/Deci-AI/super-gradients/assets/128948774/4957201b-ad43-490c-9b55-69c7184ee73c)

my torch package:
torch==2.2.0+cu118
torchaudio==2.2.0+cu118
torchmetrics==0.8.0
torchvision==0.17.0+cu118

Pytorch available GPU : NVIDIA GeForce GTX 1650

### Versions

![Screenshot 2024-02-21 115854](https://github.com/Deci-AI/super-gradients/assets/128948774/8b1388fa-bd47-4f37-9f95-7e968c9ebdf2)
![Screenshot 2024-02-21 120043](https://github.com/Deci-AI/super-gradients/assets/128948774/5f7bba3a-2de2-4f4b-ad7a-eeb40db74d99)

**Comment by BloodAxe** [comment]
Probably related issue https://github.com/Deci-AI/super-gradients/issues/1838
May I ask what SG version are you using?

**Comment by akashpat1098** [comment]
I am also having the same problem when I am using cuda,  no bounding box is displayed and but with cpu , it is working normally.
Please provide us a solution.

**Comment by BloodAxe** [comment]
To provide you a solution @akashpat1098 we need to reproduce the issue. So far I cannot reproduce the reported issue neither in Colab environment nor locally.

**Comment by BloodAxe** [comment]
Just in case - can you do `pprint(yolo_nas_pose.to('cuda').predict("3.jpg").prediction)` ?

**Comment by RamesChan** [comment]
There are no results. All when running this code:
` print(yolo_nas_pose.to('cuda').predict("3.jpg").prediction)`
But when I don't use the print() function, the output can be displayed but is blank when using the GPU.
![Screenshot 2024-02-22 092044](https://github.com/Deci-AI/super-gradients/assets/128948774/eef7f7fb-055e-4c8e-8a03-223dc2f668bb)

When I did the same experiment with CPU usage, I found that when using the print() function, the results didn't come out either.
The results appear when print() is not used and of course the results are obtained from the model as shown in the picture.
![Screenshot 2024-02-22 092404](https://github.com/Deci-AI/super-gradients/assets/128948774/b9c5bc86-95ab-4cc4-8f66-d7ec40446a48)

**Comment by VisionDan** [comment]
Hi! 
Do you have installed the torchvision cuda version? 
![grafik](https://github.com/Deci-AI/super-gradients/assets/128218314/15d2691f-84d1-45f6-81e3-054693aea2ce)

**Comment by M3nxudo** [comment]
I think this is the same issue I was having on #1834

**Comment by BloodAxe** [comment]
Could be. I was not able to reproduce this issue. Tested on 4090 and 1070 and code from [1834](https://github.com/Deci-AI/super-gradients/issues/1834#issuecomment-1945969951) works well on both GPUs.
Can you please attach `print(torch.cuda.get_device_capability(device))` for your GPU? They output may look like this: `(6, 1)`.

**Comment by M3nxudo** [comment]
This is the output I get from running the above command:
`(7, 5)`

**Comment by BloodAxe** [comment]
We have a workaround [PR](https://github.com/Deci-AI/super-gradients/pull/1881) to disable mixed precision used in `model.predict()` which hopefully should fix your issue. This will land in a next release of SG.
But if you are really eager to try it out you can install development version of SG from feature branch using this command:
`pip install -U git+https://github.com/Deci-AI/super-gradients@feature/SG-000-introduce-fp16-flag-to-predict`

**Comment by M3nxudo** [comment]
Just tested on the code provided in #1834
By changing line 24 of my code to:
`    resultsgpu = modelgpu.predict(img, fp16=False)`
Everything is working as expected now (detections on gpu are finally up and running)

**Comment by RamesChan** [comment]
> This is the output I get from running the above command: `(7, 5)`

**Comment by RamesChan** [comment]
Your solution worked. I can now use yolo-nas on cuda as well by adding fp16=False. Thank you very much. @M3nxudo @BloodAxe
